scipy/spatial/transform/_rotation.pyx

```diff
@@ -2806,54 +2833,237 @@ cdef class Rotation:
+        if n_inf > 1:
+            raise ValueError("Only one infinite weight is allowed")
+
+        elif n_inf == 1:
+            if return_sensitivity:
+                raise ValueError("Cannot return sensitivity matrix with an "
+                                 "infinite weight or one vector pair")
+            a_pri = a[weight_is_inf]
+            b_pri = b[weight_is_inf]
+            a_pri /= _norm3(a_pri[0])
+            b_pri /= _norm3(b_pri[0])
+
+            # We first find the minimum angle rotation between the primary
+            # vectors.
+            cross = np.cross(b_pri[0], a_pri[0])
+            theta = atan2(_norm3(cross), np.dot(a_pri[0], b_pri[0]))
+            if theta < 1e-3:  # small angle Taylor series approximation
+                theta2 = theta * theta
+                r = cross * (1 + theta2 / 6 + theta2 * theta2 * 7 / 360)
+            else:
+                r = cross * theta / np.sin(theta)
+            R_pri = cls.from_rotvec(r)
+
+            if N == 1:
+                # While the else clause below will work for a single pair of
+                # vectors (~weight_is_inf is empty and phi will be 0), it's
+                # more efficient to skip those calculations and just return the
+                # primary rotation here
+                R_opt = R_pri
+            else:
+                a_non_inf = a[~weight_is_inf]
+                b_non_inf = b[~weight_is_inf]
+                weights_non_inf = weights[~weight_is_inf]
+
+                # We apply the first rotation to the b vectors to align the
+                # primary vectors, resulting in vectors c. The secondary
+                # vectors must now be rotated about that primary vector to best
+                # align c to a.
+                c_non_inf = R_pri.apply(b_non_inf)
+
+                # Calculate vector components for the angle calculation. The
+                # angle phi to rotate a single 2D vector C to align to 2D
+                # vector A in the xy plane can be found with the equation
+                # phi = atan2(cross(C, A), dot(C, A)).
+                # The below equations perform the same operation, but with the
+                # 3D rotation restricted to the plane normal to a_pri.
+                # Note that einsum('ij,ij->i', X, Y) is the row-wise dot
+                # product of X and Y.
+                S = np.einsum('ij,ij->i',
+                              np.cross(c_non_inf, a_non_inf), a_pri)
+                T = (np.einsum('ij,ij->i', c_non_inf, a_non_inf)
+                     - (np.einsum('ij,ij->i', c_non_inf, a_pri)
+                        * np.einsum('ij,ij->i', a_non_inf, a_pri)))
+
+                # The weighted sum of the components essentially gives the
+                # average of all the rotation angles for each secondary
+                # vector pair. This average is the rotation angle about the
+                # primary axis that best aligns all the secondary vectors
```

Comment:
Well, if possible try to combine this and the previous comment to make it more clear.

I think that just saying that we solve a 2D alignment problem is mostly sufficient.

By the way, this approach of finding the mean direction (angle) is used in circular statistics, where the average sine and cosine is found first.

Reply:
Consolidated these comments a bit. I do think having some exposition here is useful for whoever else reads this code, since I think vector math can be pretty inscrutable if presented without context.